src/controllers/users.go

```diff
@@ -0,0 +1,169 @@
+	userOutputs := types.UsersOutput{}
+	u.logger.Debug(fmt.Sprintf("Trying to get all rolebindings in %q namespace", namespace))
+
+	roleBindings, err := u.client.RbacV1().RoleBindings(namespace).List(u.ctx, metav1.ListOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could not get rolebindings with error: %v", err.Error()))
+		return types.UsersOutput{}, err
+	}
+	u.logger.Debug("Fetched all rolebindings successfully")
+
+	for _, roleBinding := range roleBindings.Items {
+		userOutputs.Users = append(userOutputs.Users, types.User{Name: roleBinding.Name, Role: convertToPlatformRole(roleBinding.RoleRef.Name)})
+	}
+	userOutputs.Count = len(roleBindings.Items)
+
+	return userOutputs, nil
+}
+
+func (u *userController) GetUser(userIdentifier types.UserIdentifier) (types.User, error) {
+	userOutput := types.User{}
+	u.logger.Debug(fmt.Sprintf("Trying to fetch rolebinding %q in %q namespace", userIdentifier.UserName, userIdentifier.NamespaceName))
+
+	roleBinding, err := u.client.RbacV1().RoleBindings(userIdentifier.NamespaceName).Get(u.ctx, userIdentifier.UserName, metav1.GetOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could fetch rolebinding %q with error: %v", userIdentifier.UserName, err.Error()))
+		return userOutput, err
+	}
+	u.logger.Debug(fmt.Sprintf("fetched roleBinding %q successfully", userIdentifier.UserName))
+	userOutput.Name = roleBinding.Name
+	userOutput.Role = convertToPlatformRole(roleBinding.RoleRef.Name)
+
+	return userOutput, nil
+}
+
+func (u *userController) AddUser(user types.UserInput) (types.User, error) {
+	userOutput := types.User{}
+	u.logger.Debug(fmt.Sprintf("Trying to create rolebinding %q in %q namespace", user.Name, user.Namespace))
+
+	roleBinding, err := u.client.RbacV1().RoleBindings(user.Namespace).Create(u.ctx,
+		prepareRoleBinding(user.Name, user.Role), metav1.CreateOptions{})
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could create rolebinding %q with error: %v", user.Name, err.Error()))
+		return userOutput, err
+	}
+	u.logger.Debug(fmt.Sprintf("created roleBinding %q successfully", user.Name))
+	userOutput.Name = roleBinding.Name
+	userOutput.Role = user.Role
+
+	return userOutput, nil
+}
+
+func (u *userController) UpdateUser(user types.UserInput) (types.User, error) {
+	userOutput := types.User{}
+
+	u.logger.Debug(fmt.Sprintf("Trying to update rolebinding %q in %q namespace", user.Name, user.Namespace))
+
+	_, err := u.DeleteUser(types.UserIdentifier{UserName: user.Name, NamespaceName: user.Namespace})
+	if err != nil {
+		return userOutput, err
+	}
+
+	err = retry.OnError(retry.DefaultRetry, func(err error) bool {
+		return err != nil
+
+	}, func() error {
+		_, err := u.AddUser(user)
+		return err
+	})
+
+	if err != nil {
+		u.logger.Error(fmt.Sprintf("Could not update rolebinding %q with error: %v", user.Name, err.Error()))
+		return userOutput, err
+	}
+
+	u.logger.Debug(fmt.Sprintf("updated roleBinding %q successfully", user.Name))
```

Comment:
Is it possible to get `name` from an updated object, and not from the request parameters?